lib/iris/mesh/components.py

```diff
@@ -2928,6 +3071,10 @@ def copy(self, points=None, bounds=None):
         new_coord = MeshCoord(mesh=self.mesh, location=self.location, axis=self.axis)
         return new_coord
 
+    def rename(self, name: str | None) -> None:
+        """Ensure you that renaming MeshCoords is done via the Coord attached to the Mesh."""
```

Comment:
A more accurate description would be:

"""Raises an error.

Standard renaming is overriden to ensure that...